data/tools/wesnoth/wmlparser3.py

```diff
@@ -371,11 +373,15 @@ def parse_binary(self, binary: bytes, defines="") -> RootNode:
         """
         Parse a chunk of binary WML.
         """
-        temp = tempfile.NamedTemporaryFile(prefix="wmlparser_",
-                                           suffix=".cfg")
+        global tmpfiles_to_clean
```

Comment:
I dont think this line is needed, you do not assign value to that variable here.

Reply:
At line 384 I append a value to this list. Where should I put the "global" statement then?

Reply:
Not needed - you do not assign value. For appending you query the variable only - get the memory reference, and modify that underlying data structure.

Simple example
```
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = []
>>> def f(x):
...     a.append(x)
... 
>>> f(1)
>>> f(2)
>>> a
[1, 2]
>>> 

```
and
```
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = 1
>>> def f(x):
...     a = x
... 
>>> f(2)
>>> a
1
>>> 

```

Reply:
Ok true you are right, I actually had to put it because in the first implementation I didn't use a list but a single string just to test if the created file was deleted. Thanks for the correction. I updated the pull request.